### PR TITLE
samples: microbit:  update for gpio_dt_spec & DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
@anangl agreed with your comments about not needing device_is_ready() twice for pong/sound.  However, I don't see how I can use a `BUILD_ASSERT`

I tried something like:

`BUILD_ASSERT((sw0_gpio.port == sw1_gpio.port), "SW0 and SW1 need to be on same gpio controller");`

and got:

```
/Users/galak/git/zephyr/samples/boards/bbc_microbit/sound/src/main.c:33:29: error: expression in static assertion is not constant
   33 | BUILD_ASSERT((sw0_gpio.port == sw1_gpio.port), "SW0 and SW1 need to be on same gpio controller");
      |              ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/Users/galak/git/zephyr/include/zephyr/toolchain/gcc.h:77:51: note: in definition of macro 'BUILD_ASSERT'
   77 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
```

**Comment by galak** [comment]
> @anangl agreed with your comments about not needing device_is_ready() twice for pong/sound. However, I don't see how I can use a `BUILD_ASSERT`
> 
> I tried something like:
> 
> ```
> static const struct gpio_dt_spec sw0_gpio = GPIO_DT_SPEC_GET(DT_ALIAS(sw0), gpios);
> static const struct gpio_dt_spec sw1_gpio = GPIO_DT_SPEC_GET(DT_ALIAS(sw1), gpios);
> 
> BUILD_ASSERT((sw0_gpio.port == sw1_gpio.port), "SW0 and SW1 need to be on same gpio controller");
> ```
> 
> and got:
> 
> ```
> /Users/galak/git/zephyr/samples/boards/bbc_microbit/sound/src/main.c:33:29: error: expression in static assertion is not constant
>    33 | BUILD_ASSERT((sw0_gpio.port == sw1_gpio.port), "SW0 and SW1 need to be on same gpio controller");
>       |              ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
> /Users/galak/git/zephyr/include/zephyr/toolchain/gcc.h:77:51: note: in definition of macro 'BUILD_ASSERT'
>    77 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
>       |                                                   ^~~~
> ```

nevermind, remembered that we have `DT_SAME_NODE` and can use that.

**Review by anangl — APPROVED** [review]

File: samples/boards/bbc_microbit/sound/src/main.c [hunk 1]
```diff
@@ -22,7 +22,8 @@
 
 #define US_TO_HZ(_us)  (USEC_PER_SEC / (_us))
 
-static const struct device *pwm;
+static const struct device *pwm = DEVICE_DT_GET_ONE(nordic_nrf_sw_pwm);
```
Comment (gmarull) [inline samples/boards/bbc_microbit/sound/src/main.c:25]: let's use `pwm_dt_spec`

File: samples/boards/bbc_microbit/pong/src/main.c [hunk 2]
```diff
@@ -110,7 +110,7 @@ static int64_t b_timestamp;
 #define SOUND_PERIOD_PADDLE  200
 #define SOUND_PERIOD_WALL    1000
 
-static const struct device *pwm;
+static const struct device *pwm = DEVICE_DT_GET(DT_INST(0, nordic_nrf_sw_pwm));
```
Comment (gmarull) [inline samples/boards/bbc_microbit/pong/src/main.c:113]: ditto

File: samples/boards/bbc_microbit/sound/src/main.c (hunk 1 above)
Comment (gmarull) [reply samples/boards/bbc_microbit/sound/src/main.c:25]: if sample doesn't use DT it needs to be fixed, I can do that later

File: samples/boards/bbc_microbit/line_follower_robot/src/main.c [hunk 3]
```diff
@@ -11,7 +11,6 @@
 #include <zephyr/device.h>
 
 #define I2C_SLV_ADDR 0x10
-#define I2C0_LABEL DT_LABEL(DT_NODELABEL(i2c0))
 #define EXT_P13_GPIO_PIN 23     /* P13, SPI1 SCK */
 #define EXT_P14_GPIO_PIN 22     /* P14, SPI1 MISO */
```
Comment (gmarull) [inline samples/boards/bbc_microbit/line_follower_robot/src/main.c:15]: will fix in a follow-up PR, this needs to be in DT

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**